### PR TITLE
[Py3] allow to run wsgi - part1

### PR DESCRIPTION
**Pull request by MartinBasti**
With these patches we can run commands with server running on py3

Note: to use py3 install module `python3-mod_wsgi` that enables py3 wsgi automatically

Note: this may or may not depend (I haven't tested) on my PR #382 so it contains all patches, will be rebased when PR #382 merged

first WSGI related commit is 
- py3: session.py decode server name to str ￼…			7ff4b83

### CODE REVIEW

**Comment by tiran** [comment]
@MartinBasti cert tests are failing. I have restarted the failing job. Let's see if the error persists or was just caused by a Travis hick up.

**Comment by MartinBasti** [comment]
@tiran we found the issue that caues random test fails, @HonzaCholasta will provide PR with fix, that should be pushed before these commits

**Comment by MartinBasti** [comment]
Rebased, ready to be reviewed

File: ipaserver/session.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 import os
 import re
 import time
+import io
```
Comment (tiran) [inline ipaserver/session.py]: Is there any particular reason you are using the io module here? In Python 3 ```io.open is open```. In Python 2 the io module's main use is to auto-encode UTF-8 text. It's not wrong, I'm just curious.

File: ipalib/x509.py [hunk 2]
```diff
@@ -85,9 +85,16 @@ def strip_header(pem):
     """
     Remove the header and footer from a certificate.
     """
-    s = pem.find("-----BEGIN CERTIFICATE-----")
+    if isinstance(pem, bytes):
```
Comment (tiran) [inline ipalib/x509.py]: How about a regular expression?
```
def strip_header(pem):
    regexp = u"^-----BEGIN CERTIFICATE-----\s+(.*?)\s+^-----END CERTIFICATE-----"
    if isinstance(pem, bytes):
        regexp = regexp.encode('ascii')
    s = re.search(regexp, pem, re.MULTILINE | re.DOTALL)
    if s is not None:
        return s.group(1)
    else:
        return pem
```

File: ipaserver/session.py (hunk 1 above)
Comment (MartinBasti) [inline ipaserver/session.py]: I wanted to use the same

File: ipalib/x509.py (hunk 2 above)
Comment (MartinBasti) [inline ipalib/x509.py]: Good idea, thanks

**Comment by MartinBasti** [comment]
Fixed upstream
master:
https://fedorahosted.org/freeipa/changeset/a9fec1de1aa2b3c0f4c4ec6eff25ff2e75c774b0
https://fedorahosted.org/freeipa/changeset/9739d0354a8ac5fd357f7d131b3f75aa05df058b
https://fedorahosted.org/freeipa/changeset/7e8eb533752bbf5e2f05ec6bfb0ffefa8e9dcddf
https://fedorahosted.org/freeipa/changeset/35e135c4e3a7f0bf21ed4c838b8f76b43701a047
https://fedorahosted.org/freeipa/changeset/cca9aa43e146f15e235eee1197209d0ca88eb39c
https://fedorahosted.org/freeipa/changeset/aa036e5f332ef0b1ebbff6b824e236b1eeaf076e
https://fedorahosted.org/freeipa/changeset/dd3d9f1ca61946ea5d7daa17ba1d8a883922d526
https://fedorahosted.org/freeipa/changeset/49333058c869dd4bd654a7974e6e144ffd3f0dc3
https://fedorahosted.org/freeipa/changeset/b37d18288d40b4ec0b5a8df676456e09ae5f26c1
https://fedorahosted.org/freeipa/changeset/deaf9ae2473833dacb64c4961db3ae9f7c570ebd
https://fedorahosted.org/freeipa/changeset/1023cfebff99af165212dee94290a05754297270
https://fedorahosted.org/freeipa/changeset/d5ab0637fe89cbcb61491fe08b7376aeaf7ccdb8
https://fedorahosted.org/freeipa/changeset/47e76e16ef2e5d714881f3cce204611a95b4e5c8
https://fedorahosted.org/freeipa/changeset/b8d6524d43dd0667184aebc79fb77a9b8a46939a
https://fedorahosted.org/freeipa/changeset/980c8a5f9e4ccbcd3c11def9cab33d0e61e945ae

**Pull request closed without merging**